### ISSUE TITLE
Can not use map on ElasticquentResultCollection

### ISSUE THREAD

**Issue by onbjerg** [elasticquent/Elasticquent#47]
This code will not work:

```php
$books = Book::search('moby dick');
$titles = $books->map(function ($book) {
    return $book->title;
});
```

This is the error that is thrown:

```
PHP warning:  Missing argument 2 for Elasticquent\ElasticquentResultCollection::__construct(), called in /app/bootstrap/cache/compiled.php on line 12523 and defined in /app/vendor/elasticquent/elasticquent/src/ElasticquentResultCollection.php on line 22
```

**Comment by onbjerg** [comment]
The "fix" in #6 works for this one, too:

 ```php
use  \Illuminate\Database\Eloquent\Collection;

$results = new Collection(Book::search('moby dick'));
$titles = $results->map(function ($book) {
    return $book->title;
});
```

Probably also works for #12.

**Comment by onbjerg** [comment]
Ah! It's due to the fact that we require the ``$instance`` parameter in the constructor. Somehow this needs to be circumvented. The methods create a new collection (see [here for an example](https://github.com/laravel/framework/blob/5.2/src/Illuminate/Support/Collection.php#L505))

**Comment by onbjerg** [comment]
A fix (albeit an ugly one) would be to expose a method that sets the instance the collection should use to form items in the collection and get rid of the parameter in the constructor.

This would fix #47, #48, #12 and #6 (+ you could get rid of the chunk method you have implemented). Thoughts?

**Comment by timgws** [comment]
This sounds like a viable solution.

I really want to create a test suite soon so that we can actually catch issues like this and ensure that all the problems (such as the mentioned #47, #48, #12 and #6) are actually all fixed.

**Comment by onbjerg** [comment]
Fixed in #49
